src/backend/distributed/utils/maintenanced.c

```diff
@@ -345,6 +366,42 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 
 
 /*
+ * MetadataTableAccessible returns whether the given metadata table exists
+ * and can be locked using an AccessShareLock.
+ */
+static bool
+MetadataTableAccessible(const char *tableName)
+{
+	bool tableAccessible = false;
+
+	Oid relationId = get_relname_relid(tableName, PG_CATALOG_NAMESPACE);
+	if (OidIsValid(relationId))
+	{
+		bool lockAcquired = false;
+
+		lockAcquired = ConditionalLockRelationOid(relationId, AccessShareLock);
+		if (lockAcquired)
+		{
+			/*
+			 * We obtained the lock, but the relation may have just been dropped.
```

Comment:
Is that really possible? I think after the metadata clear, we shouldn't need that? How could someone drop a table if we're holding a lock?
I'm planning to remove this part if you're OK? (I also tested without this, and all looks good)